adapters/excon/lib/opentelemetry/adapters/excon/middlewares/tracer_middleware.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Excon
+      module Middlewares
+        # Excon middleware for instrumentation
+        class TracerMiddleware < ::Excon::Middleware::Base
+          HTTP_METHODS_SYMBOL_TO_STRING = {
+            connect: 'CONNECT',
+            delete: 'DELETE',
+            get: 'GET',
+            head: 'HEAD',
+            options: 'OPTIONS',
+            patch: 'PATCH',
+            post: 'POST',
+            put: 'PUT',
+            trace: 'TRACE'
+          }.freeze
+
+          def request_call(datum) # rubocop:disable Metrics/MethodLength
+            begin
+              unless datum.key?(:otel_span)
+                http_method = HTTP_METHODS_SYMBOL_TO_STRING[datum[:method]]
+
+                tracer.start_span(
+                  "HTTP #{http_method}",
+                  attributes: {
+                    'component' => 'http',
+                    'http.host' => datum[:host],
+                    'http.method' => http_method,
+                    'http.scheme' => datum[:scheme],
+                    'http.target' => datum[:path]
+                  },
+                  kind: :client
+                ).tap do |span|
+                  datum[:otel_span] = span
+                  propagator.inject(span.context, datum[:headers])
```

Comment:
Since the context prop PR merged this should be:

```suggestion
                  tracer.with_span(span) do
                    OpenTelemetry.propagation.inject(datum[:headers])
                  end
```

The reason for `with_span` is to make this span the current span in the current context. `in_span` will do that automatically, but `start_span` doesn't. `in_span` is not a good fit for this situation because it will finish the span after executing the block.